api/v1_users_verify_token.go

```diff
@@ -0,0 +1,162 @@
+package api
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"bridgerton.audius.co/api/dbv1"
+	"bridgerton.audius.co/trashid"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/gofiber/fiber/v2"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+type VerifyTokenQueryParams struct {
+	Token string `query:"token" validate:"required"`
+}
+
+func (app *ApiServer) v1UsersVerifyToken(c *fiber.Ctx) error {
+	queryParams := VerifyTokenQueryParams{}
+	if err := app.ParseAndValidateQueryParams(c, &queryParams); err != nil {
+		return err
+	}
+
+	// 1. Break JWT into parts
+	tokenParts := strings.Split(queryParams.Token, ".")
+	if len(tokenParts) != 3 {
+		return fiber.NewError(fiber.StatusBadRequest, "Invalid JWT token format")
+	}
+
+	base64Header := tokenParts[0]
+	base64Payload := tokenParts[1]
+	base64Signature := tokenParts[2]
+
+	// 2. Decode the signature
+	// Add padding if needed for base64 decoding
+	paddedSignature := base64Signature
+	if len(paddedSignature)%4 != 0 {
+		paddedSignature += strings.Repeat("=", 4-len(paddedSignature)%4)
+	}
+
+	signatureDecoded, err := base64.URLEncoding.DecodeString(paddedSignature)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "The JWT signature could not be decoded")
+	}
+
+	// Convert decoded bytes to string (hex string) and then decode hex to bytes
+	signatureHex := string(signatureDecoded)
+	signatureBytes := common.FromHex(signatureHex)
+
+	// 3. Recover wallet from signature using ethereum message recovery
+	message := fmt.Sprintf("%s.%s", base64Header, base64Payload)
+
+	// Create the ethereum signed message format
+	encodedToRecover := []byte(message)
+	prefixedMessage := []byte(fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(encodedToRecover), encodedToRecover))
+	finalHash := crypto.Keccak256Hash(prefixedMessage)
+
+	if len(signatureBytes) != 65 {
+		return fiber.NewError(fiber.StatusBadRequest, "The JWT signature could not be decoded")
```

Comment:
Any harm in making these a _little_ more specific so that errors map 1:1 with lines of code? Or so requesting client knows why the JWT couldn't be decoded.